### PR TITLE
feat: Add kroki.io Embed Service, add self-hosted kroki

### PR DESCRIPTION
**Pull request by WarlockFish**
Add the kroki.io embed 
image

### CODE REVIEW

File: app/scenes/Settings/SelfHosted.tsx [hunk 1]
```diff
@@ -98,6 +121,21 @@ function SelfHosted() {
           />
         </SettingRow>
 
+        <SettingRow
+          label={t("Kroki.self deployment")}
+          name="krokiSelfUrl"
+          description={t(
+            "Add your self-hosted kroki.io installation url here to enable automatic creates diagrams from textual descriptions!."
```
Comment (tommoor) [inline app/scenes/Settings/SelfHosted.tsx:128]:
```suggestion
            "Add your self-hosted kroki.io installation url here to enable automatic embedding of diagrams from textual descriptions."
```

File: shared/editor/embeds/Kroki.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
+import Frame from "../components/Frame";
+import Image from "../components/Img";
+import { EmbedProps as Props } from ".";
+
+function Kroki(props: Props) {
+  const embedUrl = props.attrs.matches[0];
+  const params = new URL(embedUrl).pathname.split("/");
+  const titlePrefix = new URL(embedUrl).hostname;
+  const imgGiagram = params[1];
```
Comment (tommoor) [inline shared/editor/embeds/Kroki.tsx:10]: What is "Giagram"?

File: shared/editor/embeds/Kroki.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      canonicalUrl={props.attrs.href}
+      title={title}
+      width="100%"
+      scrolling="auto"
+      border
+    />
+  );
+}
+
+// Kroki.ENABLED = [new RegExp("https?://kroki.io/(.*)/(.*)/(.*)")];
```
Comment (tommoor) [inline shared/editor/embeds/Kroki.tsx:30]:
```suggestion
```

File: shared/editor/embeds/Kroki.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Kroki.ENABLED = [/^https:\/\/kroki\.io\/[a-zA-Z]+\/[a-zA-Z]+\/.*/];
+
+Kroki.URL_PATH_REGEX = /\/[a-zA-Z]+\/[a-zA-Z]+\/.*/;
```
Comment (tommoor) [inline shared/editor/embeds/Kroki.tsx:34]:
```suggestion
```

**Review by tommoor — CHANGES_REQUESTED** [review]
Seems like a good base 👍

File: shared/editor/embeds/Kroki.tsx (hunk 2 above)
Comment (WarlockFish) [reply shared/editor/embeds/Kroki.tsx:10]: Kroki has multiple giagram analyses and  picture format ， His basic link is https://kroki.io/<plantuml>/<svg>/**， no title，so  I would like to point out the giagram analyses and  image format

Comment (tommoor) [reply shared/editor/embeds/Kroki.tsx:10]: Do you mean "diagram"? Or is this some kroki terminology

Comment (WarlockFish) [reply shared/editor/embeds/Kroki.tsx:10]: diagram  ，This is their official example ： https://kroki.io/examples.html

File: shared/editor/embeds/Kroki.tsx (hunk 4 above)
Comment (WarlockFish) [reply shared/editor/embeds/Kroki.tsx:32]: this is cp for shared/editor/embeds/Diagrams.tsx

File: app/scenes/Settings/SelfHosted.tsx [hunk 5]
```diff
@@ -98,6 +121,21 @@ function SelfHosted() {
           />
         </SettingRow>
 
+        <SettingRow
+          label={t("Kroki.io deployment")}
+          name="krokiIoUrl"
+          description={t(
+            "Add your self-hosted kroki.io installation url here to enable automatic embedding of diagrams from textual descriptions."
```
Comment (almereyda) [inline app/scenes/Settings/SelfHosted.tsx:128]:
```suggestion
            "Add your self-hosted kroki.io installation URL here to enable automatic embedding of diagrams from textual descriptions."
```

File: shared/i18n/locales/en_US/translation.json [hunk 6]
```diff
@@ -835,6 +835,8 @@
   "Allow members to create new collections within the knowledge base": "Allow members to create new collections within the knowledge base",
   "Draw.io deployment": "Draw.io deployment",
   "Add your self-hosted draw.io installation url here to enable automatic embedding of diagrams within documents.": "Add your self-hosted draw.io installation url here to enable automatic embedding of diagrams within documents.",
+  "Kroki.io deployment": "Kroki.self deployment",
+  "Add your self-hosted kroki.io installation url here to enable automatic embedding of diagrams from textual descriptions.": "Add your self-hosted kroki.io installation url here to enable automatic embedding of diagrams from textual descriptions.",
```
Comment (almereyda) [inline shared/i18n/locales/en_US/translation.json:839]:
```suggestion
  "Add your self-hosted kroki.io installation URL here to enable automatic embedding of diagrams from textual descriptions.": "Add your self-hosted kroki.io installation URL here to enable automatic embedding of diagrams from textual descriptions.",
```

**Comment by JasperSui** [comment]
When will it be merged? I think kroki.io embedding feature makes Outline more powerful.

**Comment by JasperSui** [comment]
@tommoor would you help review it again? And will it be added to www.getoutline.com as well? Thanks for the excellent work!

**Comment by WarlockFish** [comment]
Does this feature require any further modifications? Is it possible to consider merging the branch? If not, I will close it or provide some suggestions for modifying the code.

**Comment by apoorv-mishra** [comment]
Awaiting conflict resolution. I believe #5655 introduced these.

File: app/scenes/Settings/SelfHosted.tsx [hunk 7]
```diff
@@ -121,6 +149,21 @@ function SelfHosted() {
...
+          )}
+          border={false}
+        >
+          <Input
+            placeholder="https://app.kroki.io/"
```
Comment (apoorv-mishra) [inline app/scenes/Settings/SelfHosted.tsx:161]:
```suggestion
            placeholder="https://kroki.io/"
```

**Comment by apoorv-mishra** [comment]
Looks like you haven't clearly resolved the older comments. You'd need to fix those, just marking resolved won't suffice.

**Comment by santiagobiali** [comment]
I would like to see this feature implemented into outline. +1

**Pull request closed without merging**